packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx

```diff
@@ -78,7 +86,11 @@ const SignInMethodsForm = () => {
                 value: method,
                 title: t('sign_in_exp.sign_in_methods.methods', { context: method }),
               }))}
-              onChange={onChange}
+              onChange={(value) => {
+                const oldPrimaryMethod = getValues('signInMethods.primary');
+                onChange(value);
+                postPrimaryMethodChange(oldPrimaryMethod, getValues('signInMethods.primary'));
```

Comment:
`value` is the new primary value, isn't it? You don't have to call getValues again

Reply:
This is used to avoid type casting.